### PR TITLE
[GEOS-10726] ogc-api: add encoding tag with charset taken from geoserver settings

### PR DESCRIPTION
**Pull request by NielsCharlier**
[GEOS-10726](https://osgeo-org.atlassian.net/browse/GEOS-10726)

  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: src/community/ogcapi/ogcapi-core/src/main/resources/org/geoserver/ogcapi/common-header.ftl [hunk 1]
```diff
@@ -2,6 +2,7 @@
 <!DOCTYPE html>
 <html lang="en">
   <head>
+      <meta http-equiv="content-type" content="text/html; charset=${.output_encoding}">
```
Comment (aaime) [inline src/community/ogcapi/ogcapi-core/src/main/resources/org/geoserver/ogcapi/common-header.ftl:5]: There is a fixed ``<meta charset="UTF-8">`` [a few lines below](https://github.com/geoserver/geoserver/blob/main/src/community/ogcapi/ogcapi-core/src/main/resources/org/geoserver/ogcapi/common-header.ftl#L10
)... not sure what happens if there are two, but if you did manual testing, that could have interfered.

File: src/community/ogcapi/ogcapi-tiled-features/src/main/java/org/geoserver/ogcapi/features/tiled/TiledFeaturesExtension.java [hunk 2]
```diff
@@ -59,20 +60,20 @@ public TiledFeaturesExtension(FreemarkerTemplateSupport templateSupport) {
     }
 
     @Override
-    public String getExtension(Request dr, Map<String, Object> model, List htmlExtensionArguments)
+    public String getExtension(Request dr, Map model, Charset charset, List htmlExtensionArguments)
```
Comment (aaime) [inline src/community/ogcapi/ogcapi-tiled-features/src/main/java/org/geoserver/ogcapi/features/tiled/TiledFeaturesExtension.java:63]: Why was ``Map<String, Object>`` relaxed backj to ``Map``?

**Review by aaime — COMMENTED** [review]
Overall does not look bad, but some changes look strange... it seems like a change that was done on an older branch and somehow forward ported to main?

**Comment by NielsCharlier** [comment]
Indeed, this still needed updating. I fixed it

**Pull request merged**